File: atlas/c_example/example_blas_level_1.c
#include <stdio.h>
#include "cblas.h"

#define N 4

int main()
{
  float alpha = 0.5;

  float x[N]  = {0,2,4,6};
  int incx = 1;

  float y[N] = {0,10,20,30};
  float incy = 1;

  cblas_saxpy(N, alpha, x, incx, y, incy);

  printf("%f %f %f %f\n",y[0],y[1],y[2],y[3]);

  return 0;
}

File: atlas/f_example/example_blas_level_1.f90
program main
    implicit none

    integer,parameter:: n = 4

    real(kind=4):: alpha = 0.5

    real(kind=4),dimension(n):: x  = (/0,2,4,6/)
    integer:: incx = 1

    real(kind=4),dimension(n):: y = (/0,10,20,30/)
    integer:: incy = 1

    call saxpy(N, alpha, x, incx, y, incy)

    write(*,*) y(1),y(2),y(3),y(4)
end program
